include/alpaka/event/EventGenericThreads.hpp

```diff
@@ -294,15 +295,10 @@ namespace alpaka
 
                 if(!spEventImpl->isReady())
                 {
-                    auto const enqueueCount = spEventImpl->m_enqueueCount;
+                    auto oldFuture = spEventImpl->m_future;
 
-                    // Enqueue a task that waits for the given event.
-                    queueImpl.m_workerThread.submit(
-                        [spEventImpl, enqueueCount]() mutable
-                        {
-                            std::unique_lock<std::mutex> lk2(spEventImpl->m_mutex);
-                            spEventImpl->wait(enqueueCount, lk2);
-                        });
+                    // Enqueue a task that waits for the given future of the event.
+                    queueImpl.m_workerThread.submit([oldFuture]() { oldFuture.get(); });
```

Comment:
```suggestion
                    queueImpl.m_workerThread.submit([oldFuture] { oldFuture.get(); });
```
if you want

Reply:
keep the empty `()` please, the lambdas are messy enough with them...

Reply:
I will keep `()`, I personally like it also more because it cleary shows the signature of the lambda.